src/fiat/drhook/internal/dr_hook_procinfo.F90

```diff
@@ -17,7 +17,7 @@ SUBROUTINE DR_HOOK_PROCINFO(KMYPROC, KNPROC)
 INTEGER(KIND=JPIM) :: IERROR
 
 KMYPROC=-1
-KNPROC=0
```

Comment:
This is a change in behaviour possibly with unknown consequences.
Similarly KMYPROC=-1 is an invalid value.
I would probably change get_mon_out instead to cope with this situation.

Reply:
I think you're right that this probably isn't the best place for this. However, I wonder if setting `nproc` within drhook.c is the better approach?

The call site of DR_HOOK_PROCINFO is this:
```
dr_hook_procinfo_(&myproc, &nproc);
  if (myproc < 1) myproc = 1; /* Just to enable output as if myproc was == 1 */
```
Since `myproc` is changed from `-1` to `1`, I'm wondering if we should do the same to `nproc`? Its scope is limited to within drhook.c, where it's only used to give info in outputs or `get_mon_out()`.

On a personal style preference, I think it also makes sense to have 1 process for outputs instead of 0 when there's no MPI, e.g.
`Wall-time is 0.00 sec on proc#1 (0 procs, 1 threads)`
becomes
`Wall-time is 0.00 sec on proc#1 (1 procs, 1 threads)`

Reply:
In fact it looks like the drhook.c expects that KNPROC can have negative value after calling dr_hook_procinfo_, almost as if KMYPROC and KNPROC should have intent INOUT and not change its value when MPI is not initialised. The default value for nproc is -1.
So either we should also initialise KNPROC to -1 as well, or change intent to INOUT and not change the value within DR_HOOK_PROCINFO ?
And then have logic in drhook.c to change the default if needed.